### PR TITLE
CP-2611 Rename WebSocket classes

### PR DESCRIPTION
**Pull request by evanweible-wf**
## Changes
- WSocket -> WebSocket
- WSocketException -> WebSocketException
- MockWSocket -> MockWebSocket
- Tests updated to exercise both old and new names as well as inheritance

## Testing
- [ ] CI passes (tests added/updated)

## Code Review
@trentgrover-wf 
@maxwellpeterson-wf 
@dustinlessard-wf 
@jayudey-wf 
@sebastianmalysa-wf

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Raven
Number of Findings: 0

**Comment by semveraudit-wf** [comment]
### Public API Changes
```diff
@@ line 26: package:w_transport/src/web_socket/web_socket.dart @@
+ abstract class WebSocket extends WSocket implements Stream, StreamSink {}

@@ line 15: package:w_transport/src/web_socket/web_socket_exception.dart @@
+ class WebSocketException implements Exception {}

@@ line 18: package:w_transport/src/web_socket/w_socket_exception.dart @@
- class WSocketException {}
+ class WSocketException extends WebSocketException {}

@@ line 22: package:w_transport/src/web_socket/mock/w_socket.dart @@
- abstract class MockWSocket implements WSocket {}
+ abstract class MockWSocket extends MockWebSocket implements WebSocket {}

@@ line 22: package:w_transport/src/web_socket/mock/w_socket.dart @@
- abstract class MockWSocket implements WSocket {}
+ abstract class MockWSocket extends MockWebSocket implements WebSocket {}

@@ line 17: package:w_transport/src/web_socket/w_socket_exception.dart @@
- message

@@ line 19: package:w_transport/src/web_socket/w_socket_exception.dart @@
- String toString();

@@ line 34: package:w_transport/src/web_socket/mock/w_socket.dart @@
- void onOutgoing(callback(data));

@@ line 41: package:w_transport/src/web_socket/mock/w_socket.dart @@
- void triggerServerClose([int code, String reason]);

@@ line 45: package:w_transport/src/web_socket/mock/w_socket.dart @@
- @Deprecated('in 3.0.0. Use triggerServerClose() instead.') void triggerServerError(error, [StackTrace stackTrace]);

@@ line 30: package:w_transport/src/web_socket/mock/w_socket.dart @@
- void addIncoming(data);

@@ line 34: package:w_transport/src/web_socket/mock/w_socket.dart @@
- void onOutgoing(callback(data));

@@ line 41: package:w_transport/src/web_socket/mock/w_socket.dart @@
- void triggerServerClose([int code, String reason]);

@@ line 45: package:w_transport/src/web_socket/mock/w_socket.dart @@
- @Deprecated('in 3.0.0. Use triggerServerClose() instead.') void triggerServerError(error, [StackTrace stackTrace]);

@@ line 30: package:w_transport/src/web_socket/mock/w_socket.dart @@
- void addIncoming(data);

@@ line 21: package:w_transport/src/web_socket/w_socket_exception.dart @@
- WSocketException([String this.message]);
+ WSocketException([String message]) : super(message);

@@ line 23: package:w_transport/src/web_socket/mock/w_socket.dart @@
- static Future<WSocket> connect(Uri uri, {Iterable<String> protocols, Map<String, dynamic> headers});
+ static Future<WebSocket> connect(Uri uri, {Iterable<String> protocols, Map<String, dynamic> headers});

@@ line 23: package:w_transport/src/web_socket/mock/w_socket.dart @@
- static Future<WSocket> connect(Uri uri, {Iterable<String> protocols, Map<String, dynamic> headers});
+ static Future<WebSocket> connect(Uri uri, {Iterable<String> protocols, Map<String, dynamic> headers});
```

**Comment by evanweible-wf** [comment]
@rmconsole-wf

**Comment by rmconsole-wf** [comment]
@evanweible-wf
* It looks like you have not included any tickets in the title of this PR, but this repo requires tickets for all pull requests. Please create a ticket for this change, or click <a href='https://w-rmconsole.appspot.com/api/v1/pulls/82482719/addTicketToTitle'>here</a> to have Rosie create one for you.

**Comment by maxwellpeterson-wf** [comment]
+1

File: lib/src/web_socket/mock/w_socket.dart [hunk 1]
```diff
@@ -16,113 +16,14 @@ import 'dart:async';
...
 import 'package:w_transport/src/mocks/web_socket.dart'
     show MockWebSocketInternal;
-import 'package:w_transport/src/web_socket/common/w_socket.dart';
-import 'package:w_transport/src/web_socket/w_socket.dart';
+import 'package:w_transport/src/web_socket/mock/web_socket.dart';
+import 'package:w_transport/src/web_socket/web_socket.dart';
 
-abstract class MockWSocket implements WSocket {
-  static Future<WSocket> connect(Uri uri,
+abstract class MockWSocket extends MockWebSocket implements WebSocket {
```
Comment (trentgrover-wf) [inline lib/src/web_socket/mock/w_socket.dart]: should there be a deprecation annotation here too?

Comment (evanweible-wf) [inline lib/src/web_socket/mock/w_socket.dart]: Yes, but for a different reason. In #214 I'm going to add a separate test util class for controlling the behavior of the mock web socket server, which will obviate the need for this `MockWSocket` class to be exported publicly. Then in #207 I will deprecate `MockWSocket` along with other unnecessary mock classes.

**Comment by trentgrover-wf** [comment]
+1
@jayudey-wf ready for merge

**Comment by jayudey-wf** [comment]
QA Resource Approval: +10

- [ ] Testing instruction
- [ ] Dev +1's
- [ ] Dev/QA +10 with detail of what was tested
 - passing CI
- [ ] Unit test created/updated
- [ ] All unit tests pass

Merging into master.

**Comment by evanweible-wf** [comment]
@rmconsole-wf run_merge_script

**Pull request merged**